Gui

Lukowalski97 opened the pull request.

Lukowalski97 committed:
Added basic GUI, URL browse

Lukowalski97 committed:
Added basic GUI, URL browse

Lukowalski97 committed:
minor fix with search button

Lukowalski97 committed:
minor fix with search button

Lukowalski97 committed:
Added monitoring to GUI

Lukowalski97 committed:
Merge branch 'gui' of https://github.com/klasocki/ticket-scraper into gui

Lukowalski97 committed:
minor refactor

klasocki committed:
Fix monitoring bug, refactor
Now it is not possible to have multiple threads for one event

Lukowalski97 committed:
Made classes for GUI important elements

klasocki commented on src/main/scala/gui/EventsScene.scala:
```diff
@@ -0,0 +1,105 @@
+import scraping.{LiveNationScraper, Monitor, SearchEngine}
+
+class EventsScene(private val sceneWidth: Double, private val sceneHeight: Double, private val stage: Stage
+                  , private val scraper: LiveNationScraper, private val monitor: Monitor) extends Scene(sceneWidth, sceneHeight) {
```
This comma is a bit lost :D It would be great if you could do ctrl + alt + l in IntelliJ on all files before commit, it will look better, trust me ;)

Lukowalski97 replied:
I did it and IntelliJ didn't find this :(

klasocki commented on src/main/scala/gui/EventsScene.scala:
```diff
@@ -0,0 +1,105 @@
+  textField.promptText = "concert/star/band name"
+  textField.prefWidth = textFieldWidth
+
+  val searchButton = new EventsButton("Search", offX + textFieldWidth, offYSearch)
```
This should probably be split to methods, like prepareButtons, addActionHandlers or something like that

klasocki commented on src/main/scala/gui/EventsScene.scala:
```diff
@@ -0,0 +1,105 @@
+    if (selectedEvent == null) {
+      new EventsAlert(AlertType.Error, "You must select event to monitor!", "Not selected", stage).showAndWait()
+    } else if (selectedEvent.ticketsAvailable()) {
+      new EventsAlert(AlertType.Error, "Cannot monitor event which has avalaible tickets!", "Cannot monitor", stage).showAndWait()
```
These lines are too long. If every EventsAlert has AlertType.Error, does it need to be passed in constructor?

klasocki commented on src/main/scala/gui/EventsScene.scala:
```diff
@@ -0,0 +1,105 @@
+    observeButton, getLinkButton, stopMonitorButton, resetButton)
+
+  searchButton.onAction = handle {
+    list.items = ObservableBuffer(SearchEngine.searchForEvents(scraper.getEventList, textField.getText()))
```
Searching should be done by scraper, gui shouldn't know about SearchEngine. Demeter Law, useful for the coming exam :>

klasocki commented on src/main/scala/gui/EventsScene.scala:
```diff
@@ -0,0 +1,105 @@
+    } else {
+      monitor.stopMonitoring(selectedEvent)
+      println("Stopping monitoring event:" + selectedEvent.name)
+      monitorList.items.getValue.remove(selectedEvent)
```
This could be a method in EventsList, would have better encapsulation
By the way, i think monitoredList would tell that it is a list of events that are monitored, rather than list of some monitor object

klasocki reviewed (changes requested):
It's great, but could get even more readable :D

Lukowalski97 committed:
First Refactor after code review

Refactored some code, split creating buttons and their handlers to methods, made Error default type of Alert, GUI doesn't know SearchEngine anymore, changed monitorList to monitoredEventList, but didn't took monitoring as EventsList's method, as only one instance of this class is monitoring. Still need to fix events that tickets become avalaible to be removed from monitoredEventList.

klasocki committed:
Refactor
Move ticket available check from gui to monitor
Add methods to Eventslist class
